proposals/42-human-readable-mutable-names.md

```diff
@@ -0,0 +1,235 @@
+              - adds admin related to DNS updates (need to learn DNS provider's API for automating this, which introduces soft vendor-lock-in)
+              - most likely not the best choice for security-sensitive use cases where human-readable names are not required: means DNS is used in contexts where cryptographic addressing should be used instead.
+- **OS-level DNS resolver is used for all DNS TXT record lookups.** 
+  - IPFS node is unable to control which DNS resolver is used for specific TLD, all queries go to global DNS server configured in the operating system
+    - This harms competition and slows down innovation in the decentralized naming space. For example, if user changes their DNS resolver to one that is capable of resolving all TLDs over ENS, they are unable to resolve TLDs from OpenNIC or UnstoppableDomains. 
+  - Given how DNS is implemented in OS and ISPs, queries are most likely sent in plain text and can be spoofed in LAN or MITMd at the ISP level. At the time of writing this proposal DNS over HTTPS is not supported, and it is not trivial for regular user to set up a custom DNS server to change this behavior.
+
+**TLDR:** **We need to fix and future-proof the way mutable names work in our stack:**
+- Make IPNS useful on its own, and inside of DNSLink records.
+    - DNS TXT record should be set only once, updates should be published over IPNS.
+- Future-proof the way we do human readable names by
+    - leveraging DNS protocol (not the DNS network run by ICANN) for interop with existing and future user agents and naming systems
+    - allowing flexible configuration to improve security (DoH) and interop (DNSLink)
+
+
+
+#### Assumptions &amp; hypotheses
+_What must be true for this project to matter?_
+<!--(bullet list)-->
+
+- Browser is the  main distribution channel for dapps.
+- Dapps are simply a subset of all static websites loaded from IPFS.
+- Most of websites hosted on IPFS have DNSLink set up for human-readable name.
+- We have mainstream browsers like Opera and [Brave](https://brave.com/ipfs-support/) ship support for `ipfs://` and `ipns://` URIs. 
+    - Human readable names like `ipns://en.wikipedia-on-ipfs.org` work thanks to DNSLink
+    - This removes huge UX friction for dapp developers. 
+    - Vendors like Brave are committed to [surfacing DNSLink supportin the UI](https://github.com/brave/brave-browser/issues/13609) which will improve onboarding even further.
+- Alternatives to [ICANN](https://en.wikipedia.org/wiki/ICANN)  exist, and we want them to work with our stack to decrease dependency on a single organization.
+    - [ENS](https://ens.domains) provides custom TLDs (`.eth`) and leverage DNSLink for interop with IPFS and delegated lookups with clients that are unable to run own blockchain resolver.
+        - Updating `ipfs-ns` `contenthash` is a chore and costs extra (gas etc). 
+        - Using `ipns-ns` means setting `contenthash` only once, introduces cost savings and simplifies publishing. 
```

Comment:
The convenience sense.  Question about .ENS users: are they going to be comfortable/supportive of mutating changes happening outside of the blockchain?  Haven't they now effectively lost the transaction history for part of their domain (i.e., the CID for the content the domain is pointing to)?

Reply:
ENS users have a choice between pointing at IPFS identifiers in one of two namespaces:
- `ipfs-ns` – immutable identifiers – never change
- `ipns-ns` - mutable cryptographic identifiers, only owner of the publishing key can make change and update there it points at

The latter is still a cryptographic identifier, but yes, it is being mutated outside of blockchain. 
For many that is a killer feature, because it means they don't need to pay gas every time they want to update their website. IPNS makes ENS way cheaper, more practical and I'd argue safer – you don't need to take out your hardware wallet to update your blog.